parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/ParallelConsumer.java

```diff
@@ -54,6 +55,8 @@
      */
     void poll(Consumer<ConsumerRecord<K, V>> usersVoidConsumptionFunction);
 
+    void pollBatch(int batchLevel, Consumer<List<ConsumerRecord<K, V>>> usersVoidConsumptionFunction);
```

Comment:
I think I would consider moving `specifiedBatchLevel/batchSize` from method signature to `ParallelConsumerOptions`. I think it's more natural there, possible to default, and makes it easier to add more "batch-related" config in the future (min number of records before callback invoked for example).

Reply:
mmh very good point. consider it done! any opinion on defaults?

Reply:
Hmm, unsure tbh! Maybe something like `KafkaConsumer#max.poll.records / ParallelConsumerOptions#maxConcurrency` would ensure some kind of distribution among the threads.

Not having a default and throwing IllegalStateException if `pollBatch` is called without specifying batchSize is also a viable option imo so that the user is forced to make a choice.

Reply:
>KafkaConsumer#max.poll.records / ParallelConsumerOptions#maxConcurrency 
hmm, unfortunately it's not as simple as that, as the records are pipelined heavily, with a loading factor proportional to the consumption rate by the system. Given that, I think any number I hard code would be totally arbitrary, and really is heavily if not 100% dependent on the users consuming function use case.. so I think forcing the user to choose with the exception you suggest is the way to go..